### ISSUE TITLE
Space / enter do not trigger download link

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode-docs#3778]
Reported by @jvesouza

Keyboard navigate onto a user installation `64bit` blue link or any other blue link.
Hit space or enter -> nothing happens -> I can not download vscode

![Screenshot 2020-06-11 at 16 29 33](https://user-images.githubusercontent.com/1926584/84398895-c8f99a00-ac00-11ea-853f-399bc0751ab8.png)

**Comment by gregvanl** [comment]
Hi @jvesouza, I can not reproduce your issue. If I tab through the page to the download links such as `64 bit`, pressing Enter will start the download. I have tested on Windows 10 with both the Edge and Chrome browsers.
What platform and browser are you using? Thanks

**Comment by jvesouza** [comment]
I am running on linux, firefox or chromium.

**Comment by jvesouza** [comment]
If I click in the 64 bit tar.gz down link nothing happens, but if I click in the snap store download link, the link is activated.

**Comment by gregvanl** [comment]
I am not able to repo on my Linux Ubuntu (direct install on the machine). Are you running Linux in a VM or some other setup?

**Comment by isidorn** [comment]
Nope he is running Linux straight up. @gregvanl maybe you can provide a code pointer and I can investigate more, or maybe someone on the redmond side could help? We can also look into this together tomorrow.

**Comment by Tyriar** [comment]
Enter works fine for me on Windows (I don't have a Linux box), space doesn't but it's doing what it's meant to do in Edge and scrolling down a page. @isidorn can you repro? Could this be a browser or screen reader problem?

**Comment by isidorn** [comment]
@Tyriar I can repro with no screen reader in my Linux and Win VM. I can aslo reproduce on my mac.

**Comment by Tyriar** [comment]
@isidorn what browser?

**Comment by isidorn** [comment]
@Tyriar Chrome 83.0.4103.97
I can also repro with Safari 13.1.1

**Comment by jvesouza** [comment]
> I am not able to repo on my Linux Ubuntu (direct install on the machine). Are you running Linux in a VM or some other setup?

I'm running linux on a physical machine.
The firefox version is 77.0.1 and the chromium version is 83.0.4103.97.

**Comment by gregvanl** [comment]
Can you see if it repros if you run the browser in incognito mode? That would eliminate the possibility that a browser extension is causing the problem.

**Comment by jvesouza** [comment]
@gregvanl The problem also happens when I use incognito mode.
I also restarted the FF with the extensions disabled but I was not successful.

One thing I noticed is that if I press the application key when the focus is on the link, some options like opening a link in another window and copy link location are not shown.
However these options are displayed if the focus is on the snap store link.

**Comment by Tyriar** [comment]
However these options are displayed if the focus is on the snap store link.

@jvesouza this makes sense; the snap store link is a regular link, the others are `href="#'` links so I'm not surprised if they don't tell you they're real links.

**Comment by isidorn** [comment]
Another user is also seeing this https://twitter.com/stommepoes/status/1271121224807124995

**Comment by StommePoes** [comment]
Repro'd Windows 10 on Firefox 77.0.1, Chrome 83.0.4.

This was reported by a screen reader user on the [Orca Mailing List](https://mail.gnome.org/archives/orca-list/2020-June/msg00051.html), where further testing without the screen reader showed that it's a pure keyboard issue. 

The reason for the problem is likely due to lack of `href` attribute. The link is made focusable with a `tabindex`, however links without hrefs do not have click events generated from the Enter key.
(I do have a test case from years ago but it's on a very messy page with other junk so I'm not linking to it.)
```
<a class="dlink" data-os="win64user" tabindex="0" aria-label="Windows 64 bit User Installer download link">64 bit</a>
```

Solution: either add a URL onto an `href` (assuming the resource is actually online), OR change this element to a button:
```
<button type="button" class="dlink" data-os="win64user" aria-label="Windows 64 bit User Installer download link">64 bit</button>
```
Simply changing the link element to a button will give you the click-event-generated-with-Enter-key for free (plus since this generates an action on the page rather than bringing users to a destination, it's probably a better semantic). I added the `type` simply because I do that automatically in these days of people using "components" which may be stuffed anywhere, including inside `<form>` elements.

**Comment by isidorn** [comment]
@StommePoes thanks for the awesome writeup!

**Comment by Tyriar** [comment]
This is what I see

```
<a class="dlink" data-os="win64user" href="#" role="link" aria-label="Windows User install 64 bit">64 bit</a>
```

![image](https://user-images.githubusercontent.com/2193314/84664739-d601dc00-aed3-11ea-898b-18e70ec62cb5.png)

**Comment by StommePoes** [comment]
zomgweird. So this explains why it works for @Tyriar with keyboard, but over here I don't have that at all:
![Screenshot in Edge browser with devtools open, showing no link role nor href='#' on the same 64 bit link as screenshotted in Tyriar's comment](https://user-images.githubusercontent.com/136210/84665841-9b726080-af20-11ea-9dc3-7ba5ec07554c.png)

I just tried hooking up to a US server with my VPN to see if I could get different code (in case it was some weird geo-related A/B thing), but I'm still seeing those links without `href` on my end.

**Comment by isidorn** [comment]
Yeah I also do not see the href, and yes I am not in the US
```<a class="dlink" data-os="win64user" tabindex="0" aria-label="Windows 64 bit User Installer download link">64 bit</a>```

@Tyriar might it be that we have a different version of the website deployed to different geo locations?

![Screenshot 2020-06-15 at 16 02 00](https://user-images.githubusercontent.com/1926584/84666595-91049680-af21-11ea-8f04-f12e6591ec16.png)

**Comment by Tyriar** [comment]
@gregvanl can you explain why href differs 😕

**Comment by gregvanl** [comment]
@Tyriar I can't explain why some regions don't include `href="#"`. There shouldn't be any difference in what we deploy to each regional website. I just checked the westus, eastus, and westeu sites directly and from my machine I see `href="#"` on each site.

**Comment by StommePoes** [comment]
You know, this ticket doesn't have a "how to reproduce". While the screenshots mostly all look right... what if we are looking at two different pages?
The URL I'm looking at is the Insider's page: https://code.visualstudio.com/insiders/

Are we definitely looking at the same page?

**Comment by jvesouza** [comment]
I tried to copy what the browser displays in the inspector:

`<a class="dlink" data-os="linux64" tabindex="0" aria-label="64 bit .tar.gz download link">64 bit</a>`

**Comment by Tyriar** [comment]
@StommePoes yeah I think that's it, the stable download page has `href="#"`, Insiders does not.

**Comment by isidorn** [comment]
Awesome! Sorry for not making it clear that I was downloading insiders, my mistake.

**Comment by StommePoes** [comment]
@jvesouza you absolutely did it right, that's (one of) the problematic link(s).

**Comment by gregvanl** [comment]
OK, I can repro on the Insiders download page. The initial screenshot is from the regular download page. Thanks @StommePoes for getting us "on the same page".

**Comment by gregvanl** [comment]
Fixed by adding the missing `href="#"` to the Insiders links.
Thanks to everyone for helping solve this mysterious case of mistaken identity.
